### ISSUE TITLE
Nodes become unresponsive after changing ctx.fillStyle color

### ISSUE THREAD

**Issue by turhanberk** [vasturiano/force-graph#306]
I was trying to build a custom node geometry with 2 rectangles having different colors. I directly used the "custom node shapes" example, and just modified the nodePaint function with adding another ctx.fillStyle and ctx.fillRect. 

The rectangle who takes its color from ctx.fillStyle = color is responsive where it can be dragged and labeled, where second rectangle colored with ctx.fillStyle = "#ff0000" is not. If nodePaint function does not take a single color and build the whole node with the same color, the node become unresponsive.

<img width="701" alt="Screenshot 2023-03-01 at 16 38 30" src="https://user-images.githubusercontent.com/55454990/222270695-5d67c40b-c97d-4ecd-a6ed-7afbda4e9f73.png">
This is the network I am trying to build where I cannot drag or see the labels of the nodes while I am interacting with the red rectangles located at the bottom of the each nodes. I am also adding adding a codepen to reproduce it: 

[https://codepen.io/berkturhan/pen/ZEMeJJL](https://codepen.io/berkturhan/pen/ZEMeJJL)

In the codepen notice that you cannot drag or interact with the nodes by clicking the red rectangle that is always located at the bottom, but the upper rectangle is responsive. Is there any other way to implement same structure where the whole node is draggable and responsive?

Thanks a lot!

**Comment by vasturiano** [comment]
@turhanberk thanks for reaching out.

The issue is inside your `nodePointerAreaPaint` function:
```js
({ id, x, y }, color, ctx) {
    ctx.fillStyle = color
    ctx.fillRect(x - 6, y - 4, 12, 8);
    ctx.fillStyle = "#ff0000"; // <- this line
    ctx.fillRect(x - 6 , y + 4, 12, 8);
}
```

This method is not for styling the elements but to indicate to the library their interaction area. To do this you need to style them with the supplied `color` argument, as that's the indexing key for objects in the canvas. If you use a different color it's expectable that that node will become completely unresponsive as that color is unknown to the system (or worse, may conflict with another node's unique color).

Long story short, if you need to re-style your nodes, do it inside the `nodeCanvasObject` method, not the `nodePointerAreaPaint` one.

**Comment by turhanberk** [comment]
Hey @vasturiano !

Thanks a lot for the quick reply and your help. I'll take a look at the `nodeCanvasObject` method for re-styling. 

Amazing work!
